### DISCUSSION TITLE
How does this compare to Import Linter?

### DISCUSSION THREAD

**Discussion by brianmedigate** [gauge-sh/tach#72] [Q&A] [answered]
This appears to do something similar to [Import Linter](https://github.com/seddonym/import-linter).

I was wondering if anyone has done a comparison between them, and can offer any guidance in choosing one.

Are they designed for the same use cases?
How do their feature sets compare?
Does this project have any plans/roadmap for the future which might affect the decision?

Something like this might also be useful to put in the "Why `tach`?" or "FAQ" sections of the documentation.

**Comment by caelean** [comment, marked as answer]
Hey Brian - that's a great question! I'll do a longer writeup in the future (and add it to the FAQ), but TLDR:


- Tach supports a [strict interface](https://gauge-sh.github.io/tach/strict-mode/) for packages; Import Linter does not.
- Tach automatically analyzes your existing set of imports to get you to a passing set of definitions ([tach sync](https://gauge-sh.github.io/tach/usage/#tach-sync)); Import Linter requires you to do this manually.
- The configuration and setup are very different. Import linter relies on a handwritten ruleset, while Tach utilizes an interactive cli to set itself up.
- At it's core, Import Linter operates on 'contracts', while Tach operates on 'tags'. By using tags, Tach allows you to describe one rule for many packages.

Let me know if you have any more q's! Would love any feedback you have on the tool.

**Reply by brianmedigate** [reply]
great, thanks! I'm looking forward to checking it out

**Reply by caelean** [reply]
@brianmedigate definitely let us know if you have any feedback 🙂 appreciate you checking it out!
